test/addons-napi/test_async/test.js

```diff
@@ -0,0 +1,10 @@
+'use strict';
+const common = require('../../common');
+const assert = require('assert');
+const test_array = require(`./build/${common.buildType}/test_array`);
+
+test_array(5, common.mustCall(function(err, val) {
+  assert.strictEqual(err, null);
+  assert.strictEqual(val, 10);
+  process.nextTick(common.mustCall(function() {}));
```

Comment:
I adapted this test from test/addons/async-hello-world which has this pattern. I guess they're using the empty function here to make sure that the complete callback is actually called? Good point about ```MakeCallback```. I wonder how often the ```complete``` callback would be a c++ function vs some wrapper around JavaScript?